simply/actor.py

```diff
@@ -233,3 +240,79 @@ def create_random(actor_id, start_date="2021-01-01", nb_ts=24, ts_hour=1):
     )
 
     return Actor(actor_id, df, ls=ls, ps=ps)
+
+
+def create_from_csv(actor_id, asset_dict={}, start_date="2021-01-01", nb_ts=None, ts_hour=1):
+    """
+    Create actor instance with random asset time series and random scaling factors. Replace
+
+    :param str actor_id: unique actor identifier
+    :param Dict asset_dict: nested dictionary specifying 'csv' filename and column ('col_index')
+        per Actor asset
+    :param str start_date: Start date "YYYY-MM-DD" of the DataFrameIndex for the generated actor's
+        asset time series
+    :param int nb_ts: number of time slots that should be generated, derived from csv if None
+    :param ts_hour: number of time slots per hour, e.g. 4 results in 15min time slots
+    :return: generated Actor object
+    :rtype: Actor
+    """
+    # Random scale factor generation, load and price time series in boundaries
+    ls = random.uniform(0.8, 1.3) / ts_hour
+    ps = random.uniform(1, 7) / ts_hour
+    # Probability of an actor to possess a PV, here 40%
+    pv_prob = 0.4
+    ps = random.choices([0, ps], [1 - pv_prob, pv_prob], k=1)
+
+    # Initialize DataFrame
+    cols = ["load", "pv", "schedule", "prices"]
+    df = pd.DataFrame([], columns=cols)
+
+    # Read csv files for each asset
+    csv_peak = {}
+    for col, csv_dict in asset_dict.items():
+        # if csv_dict is empty
+        if not csv_dict:
+            continue
+        csv_df = pd.read_csv(
+            csv_dict["csv"],
+            sep=',',
+            parse_dates=['Time'],
+            dayfirst=True
+        )
+        # Rename column and insert data based on dictionary
+        df.loc[:, col] = csv_df.iloc[:nb_ts, csv_dict["col_index"]]
+        # Save peak value and normalize time series
+        csv_peak[col] = df[col].max()
```

Comment:
Currently, unused dictionary

Reply:
Moved ls and ps scaling factor definition into dict and added parameter to decide if scaling factors are updated by csv peak values